frontend/src/components/explorer/mapViewer/MapsListing.vue

```diff
@@ -42,6 +42,17 @@ export default {
       showing2D: state => state.maps.showing2D,
       mapsListing: state => state.maps.mapsListing,
     }),
+    categories() {
+      const categories = [];
+      const li = Object.keys(this.mapsListing).filter(c => this.mapsListing[c].length > 0).sort();
+      for (let i = 0; i < li.length; i += 1) {
+        const category = li[i];
+        if (category !== 'customs' || this.showing2D) {
+          categories.push(category);
+        }
+      }
+      return categories;
```

Comment:
I think the following might do the same, please test it out locally though as I'm not on that branch atm.

```javascript
return Object.keys(this.mapsListing).filter(c => this.mapsListing[c].length > 0 && (c !== 'customs' || this.showing2D)).sort();
```

Reply:
@e0 Works like a charm! I'll take your suggestion for refactoring.